cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -431,6 +431,21 @@ public void deleteAll(Range keyRange) throws InvalidFieldException, IOException
     }
   }
 
+  @Override
+  public void updateAll(Range keyRange, Collection<Field<?>> fields) throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: UpdateAll with range {}", schema.getTableId(), keyRange);
+    try (CloseableIterator<StructuredRow> iterator = getFilterByRangeIterator(keyRange)) {
+      while (iterator.hasNext()) {
+        Collection<Field<?>> primaryKeys = iterator.next().getPrimaryKeys();
+        Put put = updateFieldsToBytes(primaryKeys);
```

Comment:
Is the fields to update even used?

Reply:
yeah need to update L439. I feel like there are lot of things missed without testing. Let me add the test first and then we can automatically catch these errors.